.github/workflows/release.yml

```diff
@@ -25,26 +26,28 @@ jobs:
           storage.googleapis.com:443
           uploads.github.com:443
     - name: Checkout
-      uses: actions/checkout@629c2de402a417ea7690ca6ce3f33229e27606a5
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
     - name: Set up Go 
-      uses: actions/setup-go@424fc82d43fa5a37540bae62709ddcc23d9520d4
+      uses: actions/setup-go@0aaccfd150d50ccaeb58ebd88d36e91967a5f35b # v5.4.0
       with:
         go-version: 1.24.1
     
     - name: Configure .netrc
+      env:
+        PAT: ${{ secrets.PAT }}
       run: |
         if [[ ! -e "~/.netrc" ]]; then
             touch ~/.netrc
         fi
-        printf "machine github.com login stepsecurity-infra-bot password ${{ secrets.PAT }}" >>~/.netrc
+        printf "machine github.com login stepsecurity-infra-bot password $PAT" >>~/.netrc
 
       
     - name: Create go vendor dir
       run: |
         go mod vendor
              
     
-    - uses: goreleaser/goreleaser-action@5df302e5e9e4c66310a6b6493a8865b12c555af2
+    - uses: goreleaser/goreleaser-action@9c156ee8a17a598857849441385a2041ef570552 # v6.3.0
```

Comment:
This is the latest version https://github.com/goreleaser/goreleaser-action/releases/tag/v6.3.0